core/src/main/java/hudson/model/User.java

```diff
@@ -713,8 +714,9 @@ public static boolean isIdOrFullnameAllowed(@CheckForNull String id) {
         if (id == null || StringUtils.isBlank(id)) {
             return false;
         }
+        final String trimmedId = id.trim();
         for (String invalidId : ILLEGAL_PERSISTED_USERNAMES) {
-            if (id.equalsIgnoreCase(invalidId))
+            if (trimmedId.equalsIgnoreCase(invalidId))
```

Comment:
> Specifying locale doesn't help with that either so I am not sure where you are going with this?

See my top level comment.

> if this is an issue then it needs to be fixed fully.
if it's not an issue then I see no reason for this code.

basically if users are expecting us to police usernames then we need to do it correctly - otherwise they will get used to oh System is always the System user as it is filtered, and not have the is this user X or is it User Y who has just changed his full name for display purposes....

I'm just trying to point out all the areas where you can create a user that looks like one of the internal users - but smells very different,  or that what you may think is an invalid username is actually very valid on some authentication systems.